ouroboros-consensus-diffusion/src/unstable-mock-testlib/Test/Consensus/Ledger/Mock/Generators.hs

```diff
@@ -99,14 +103,24 @@ instance Arbitrary SimpleBody where
 instance Arbitrary (SomeSecond (NestedCtxt Header) (SimpleBlock c ext)) where
   arbitrary = return $ SomeSecond indexIsTrivial
 
-instance Arbitrary (SomeSecond BlockQuery (SimpleBlock c ext)) where
-  arbitrary = return $ SomeSecond QueryLedgerTip
+instance Arbitrary (SomeBlockQuery (BlockQuery (SimpleBlock c ext))) where
+  arbitrary = return $ SomeBlockQuery QueryLedgerTip
 
 instance (SimpleCrypto c, Typeable ext) => Arbitrary (SomeResult (SimpleBlock c ext)) where
   arbitrary = SomeResult QueryLedgerTip <$> arbitrary
 
-instance Arbitrary (LedgerState (SimpleBlock c ext)) where
-  arbitrary = SimpleLedgerState <$> arbitrary
+instance (SimpleCrypto c, Typeable ext)
+         => Arbitrary (LedgerState (SimpleBlock c ext) EmptyMK) where
+  arbitrary =
+        forgetLedgerTables
+    <$> arbitrary @(LedgerState (SimpleBlock c ext) ValuesMK)
```

Comment:
Should this instance generate:

* A ledger state with no ledger tables, *without* stowed UTXOs
* A ledger state with no ledger tables, *with* stowed UTXOs

Right now it's generating the former. My guess is that the latter would be more interesting

Maybe this ambiguity is an argument in favour of having two types of `EmptyMK`: `StowedEmptyMK` and `UnstowedEmptyMK`, or something along those lines

Reply:
I replaced `forgetLedgerTables` with with `stowLedgerTables`, and the `mock-test` suite passes fine :+1:

Reply:
I don't think it is worth concretizing this more until we introduce StowedMK and UnstowedMK.

I would favor names that explain not what happened with the data but instead what data is there, so something like `NothingMK` and `LedgerMK`

Reply:
#1334